app/uk/gov/hmrc/exports/scheduler/jobs/PurgeAncientSubmissionsJob.scala

```diff
@@ -0,0 +1,129 @@
+import org.mongodb.scala.model.Filters._
+import play.api.Logging
+import uk.gov.hmrc.mongo.play.json.Codecs
+import play.api.libs.json.{Format, Json}
+import uk.gov.hmrc.exports.config.AppConfig
+import uk.gov.hmrc.exports.models.declaration.ExportsDeclaration
+import uk.gov.hmrc.exports.models.declaration.notifications.{ParsedNotification, UnparsedNotification}
+import uk.gov.hmrc.exports.repositories._
+import uk.gov.hmrc.exports.models.declaration.submissions.{Submission, SubmissionRequest}
+import uk.gov.hmrc.exports.mongo.ExportsClient
+
+import java.time._
+import javax.inject.{Inject, Singleton}
+import scala.collection.convert.DecorateAsScala
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PurgeAncientSubmissionsJob @Inject() (
+  val appConfig: AppConfig,
+  submissionRepository: SubmissionRepository,
+  declarationRepository: DeclarationRepository,
+  unparsedNotificationRepository: UnparsedNotificationWorkItemRepository,
+  parsedNotificationRepository: ParsedNotificationRepository,
+  transactionalOps: PurgeSubmissionsTransactionalOps
+)(implicit ec: ExecutionContext)
+    extends ScheduledJob with ExportsClient with Logging with DecorateAsScala {
+
+  lazy val submissionCollection = db.getCollection(submissionRepository.collectionName)
+  lazy val declarationCollection = db.getCollection(declarationRepository.collectionName)
+  lazy val notificationCollection = db.getCollection(parsedNotificationRepository.collectionName)
+  lazy val unparsedNotificationCollection = db.getCollection(unparsedNotificationRepository.collectionName)
+
+  override val name: String = "PurgeAncientSubmissions"
+
+  override def interval: FiniteDuration = jobConfig.interval
+  override def firstRunTime: Option[LocalTime] = Some(jobConfig.elapseTime)
+
+  private def jsonSettings(mode: JsonMode) = JsonWriterSettings.builder.outputMode(mode).build
+
+  private val jobConfig = appConfig.purgeAncientSubmissions
+  private val clock: Clock = appConfig.clock
+
+  private val latestStatus = "latestEnhancedStatus"
+  private val statusLastUpdated = "enhancedStatusLastUpdated"
+
+  private val expiryDate = Codecs.toBson(ZonedDateTime.now(clock).minusDays(180))
+
+  private val latestStatusLookup =
+    in(latestStatus, List("GOODS_HAVE_EXITED", "DECLARATION_HANDLED_EXTERNALLY", "CANCELLED"): _*)
+
+  private val olderThanDate = lte(statusLastUpdated, expiryDate)
+
+  override def execute(): Future[Unit] = {
+
+    implicit val formatNotification: Format[ParsedNotification] = ParsedNotification.format
+    import ExportsDeclaration.Mongo._
+
+    val submissions: List[Submission] = submissionCollection
+      .find(and(olderThanDate, latestStatusLookup))
+      .asScala
+      .flatMap {
+        parseJsonFromDocument[Submission]
+      }
+      .toList
+
+    logger.info(s"Submissions found older than 180 days: ${submissions.size}")
+
+    val declarations: List[ExportsDeclaration] = declarationCollection
```

Comment:
Why do you need to retrieve the declarations? Once you have the Submissions you can directly remove the documents